apps/api/src/modules/payments/payments.service.ts

```diff
@@ -423,16 +426,33 @@ export default class PaymentsService {
     // Attempt to find card (cardId could be source or db ID)
     const card = await PaymentCardModel.query(trx).findById(cardId)
 
+    // Temporary workaround as `localhost` is not allowed in Circle
+    const verificationHostname = Configuration.webUrl.includes('localhost')
+      ? 'https://demo.algomart.dev'
+      : Configuration.webUrl
+
     // Create payment using Circle API
     const payment = await this.circle
       .createPayment({
         idempotencyKey: idempotencyKey,
-        metadata: metadata,
+        metadata: {
+          ...metadata,
+          sessionId: SHA256(user.id).toString(enc.Base64),
+        },
         amount: {
           amount: priceInUSD,
           currency: DEFAULT_CURRENCY,
         },
-        verification: verification,
+        verification: CirclePaymentVerificationOptions.three_d_secure,
```

Comment:
Verify this line is accurate

Reply:
Add second attempt using CVV for verification when failing to create payment with 3DS.